Tests/FunctionalTests/Edge+ConsentTests.swift

```diff
@@ -0,0 +1,276 @@
+        let requestBody = getFlattenNetworkRequestBody(consentRequests[0])
+        print(requestBody)
+        XCTAssertEqual(7, requestBody.count)
+        XCTAssertNotNil(requestBody["identityMap.ECID[0].id"] as? String)
+        XCTAssertEqual("ambiguous", requestBody["identityMap.ECID[0].authenticationState"] as? String)
+        XCTAssertEqual(true, requestBody["identityMap.ECID[0].primary"] as? Bool)
+        XCTAssertEqual("Adobe", requestBody["consent[0].standard"] as? String)
+        XCTAssertEqual("2.0", requestBody["consent[0].version"] as? String)
+        XCTAssertEqual("n", requestBody["consent[0].value.collect.val"] as? String)
+        XCTAssertNotNil(requestBody["consent[0].value.metadata.time"] as? String)
+    }
+
+    func testCollectConsentYes_sendsRequestToEdgeNetwork() {
+        setExpectationNetworkRequest(url: FunctionalTestConst.EX_EDGE_CONSENT_URL_STR, httpMethod: HttpMethod.post, expectedCount: 1)
+
+        // test
+        updateCollectConsent(status: ConsentStatus.yes)
+
+        // verify
+        assertNetworkRequestsCount()
+        let interactRequests = self.getNetworkRequestsWith(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, timeout: 1)
+        XCTAssertEqual(0, interactRequests.count)
+        let consentRequests = self.getNetworkRequestsWith(url: FunctionalTestConst.EX_EDGE_CONSENT_URL_STR, httpMethod: HttpMethod.post, timeout: 1)
+        XCTAssertEqual(HttpMethod.post, consentRequests[0].httpMethod)
+        let requestBody = getFlattenNetworkRequestBody(consentRequests[0])
+        print(requestBody)
+        XCTAssertEqual(7, requestBody.count)
+        XCTAssertNotNil(requestBody["identityMap.ECID[0].id"] as? String)
+        XCTAssertEqual("ambiguous", requestBody["identityMap.ECID[0].authenticationState"] as? String)
+        XCTAssertEqual(true, requestBody["identityMap.ECID[0].primary"] as? Bool)
+        XCTAssertEqual("Adobe", requestBody["consent[0].standard"] as? String)
+        XCTAssertEqual("2.0", requestBody["consent[0].version"] as? String)
+        XCTAssertEqual("y", requestBody["consent[0].value.collect.val"] as? String)
+        XCTAssertNotNil(requestBody["consent[0].value.metadata.time"] as? String)
+    }
+
+    func testCollectConsentOtherThanYesNo_doesNotSendRequestToEdgeNetwork() {
+        // test
+        updateCollectConsent(status: ConsentStatus.pending)
+        updateCollectConsent(status: "u")
+        updateCollectConsent(status: "some value")
+
+        // verify
+        let interactRequests = self.getNetworkRequestsWith(url: FunctionalTestConst.EX_EDGE_INTERACT_URL_STR, httpMethod: HttpMethod.post, timeout: 1)
+        XCTAssertEqual(0, interactRequests.count)
+        let consentRequests = self.getNetworkRequestsWith(url: FunctionalTestConst.EX_EDGE_CONSENT_URL_STR, httpMethod: HttpMethod.post, timeout: 1)
+        XCTAssertEqual(0, consentRequests.count)
+    }
+
```

Comment:
good to have a few more tests here.. with Consents obtained from defaultConfig only and no Consent API's are called

I guess we for those test we should wait for that Bootup Consent PR to be merged.

Reply:
There are some more tests for this scenario in EdgeExtensionTests -> testBootup with different Consent shared states (see testBootup_whenNoConsentSharedState_usesDefaultYes)